hostedsms/responses.py

```diff
@@ -3,67 +3,116 @@
 """
 
 
-class HostedSmsApiResponse(object):
-    """Base class for all respones"""
-    def __init__(self, response):
-        self.current_time = response.CurrentTime
+from datetime import datetime
+from typing import Any, List, Optional
+from dataclasses import dataclass
 
+__all__ = [
+    "HostedSmsApiResponse", "DeliveryReport", "GetDeliveryReportsResponse",
+    "GetUnreadDeliveryReportsResponse", "GetReadDeliveryReportsResponse",
+    "GetValidSendersResponse", "SendSmsResponse", "SendSmsesResponse",
+    "GetInputSmsesResponse", "InputSms"
+]
 
-class DeliveryReport(object):
 
-    def __init__(self, report):
-        self.report_id = str(report.ReportId)
-        self.phone = str(report.Phone)
-        self.message_id = str(report.MessageId)
-        self.status = int(report.Status)
-        self.delivery_time = report.DeliveryTime
+class HostedSmsApiResponse:
+    """Base class for all API responses."""
+    def __init__(self, response: Any):
+        self.current_time: Any = getattr(response, 'CurrentTime', None)
 
-    def __repr__(self):
-        return '<DeliveryReport {0}>'.format(self.report_id)
 
+@dataclass
+class DeliveryReport:
+    report_id: str
+    phone: str
+    message_id: str
+    status: int
+    delivery_time: Any
 
-class GetDeliveryReportsResponse(HostedSmsApiResponse):
+    def __init__(self, report: Any):
+        self.report_id = str(getattr(report, 'ReportId', ''))
+        self.phone = str(getattr(report, 'Phone', ''))
+        self.message_id = str(getattr(report, 'MessageId', ''))
+        self.status = int(getattr(report, 'Status', 0))
+        self.delivery_time = getattr(report, 'DeliveryTime', None)
+
+    def __repr__(self) -> str:
+        return f'<DeliveryReport {self.report_id}>'
 
-    def __init__(self, response):
-        super(GetDeliveryReportsResponse, self).__init__(response)
-        self.delivery_reports = []
-        if response.DeliveryReports:
-            reports = response.DeliveryReports.DeliveryReport
-            for report in reports:
-                self.delivery_reports.append(DeliveryReport(report))
 
-    def __repr__(self):
-        return '<GetUnreadDeliveryReports {0}>'.format(self.current_time)
+class GetDeliveryReportsResponse(HostedSmsApiResponse):
+    """Response for delivery reports."""
+    def __init__(self, response: Any):
+        super().__init__(response)
+        self.delivery_reports: List[DeliveryReport] = [
+            DeliveryReport(report) for report in getattr(getattr(response, 'DeliveryReports', None), 'DeliveryReport', [])
+        ]
+
+    def __repr__(self) -> str:
+        return f'<GetUnreadDeliveryReports {self.current_time}>'
```

Comment:
```suggestion
    def __repr__(self) -> str:
        return f'<GetDeliveryReportsResponse {self.current_time}>'
```

Nazwa klasy w formatowanym stringu jest niepoprawna.